src/api/app/components/bs_request_action_source_and_target_component.rb

```diff
@@ -0,0 +1,41 @@
+class BsRequestActionSourceAndTargetComponent < ApplicationComponent
+  def initialize(bs_request)
+    super
+
+    @bs_request = bs_request
+  end
+
+  def call
+    capture do
+      if source.present?
+        concat(tag.span(source))
+        concat(tag.i(nil, class: 'fas fa-long-arrow-alt-right text-info mx-2'))
+      end
+      concat(tag.span(target))
+    end
+  end
+
+  private
+
+  def source
+    @source ||= if number_of_bs_request_actions > 1
+                  ''
+                else
+                  [bs_request_action.source_project, bs_request_action.source_package].compact.join(' / ')
+                end
+  end
+
+  def target
+    return bs_request_action.target_project if number_of_bs_request_actions > 1
+
+    [bs_request_action.target_project, bs_request_action.target_package].compact.join(' / ')
+  end
+
+  def bs_request_action
+    @bs_request_action ||= @bs_request.bs_request_actions.first
+  end
+
+  def number_of_bs_request_actions
+    @number_of_bs_request_actions ||= @bs_request.bs_request_actions.size
```

Comment:
I think, this can be set once in the constructor and then the method and the memoization is not needed if you use an `attr_reader :number_of_bs_request_actions`

Reply:
done